waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -46,9 +48,14 @@ genJobs spec = return $ genJob <$> getJobs spec
   where
     tmplFile = C.asTmplFile $ jobsDirInServerTemplatesDir SP.</> [relfile|_job.js|]
     dstFileFromJobName jobName = jobsDirInServerRootDir SP.</> fromJust (parseRelFile $ jobName ++ ".js")
+
     genJob :: (String, Job) -> FileDraft
     genJob (jobName, job) =
       let (jobPerformFnName, jobPerformFnImportStatement) = getJsImportDetailsForExtFnImport relPosixPathFromJobFileToExtSrcDir $ (J.fn . J.perform) job
+          maybeJobPerformOptions = J.performOptions . J.perform $ job
+          maybeJobSchedule =
+            J.schedule job
+              >>= (\s -> return $ object ["cron" .= J.cron s, "performFnArg" .= J.performFnArg s, "options" .= J.sheduleOptions s])
```

Comment:
I'd probably extract the lambda to a named function. Also, it looks like we don't need to use the monadic bind (fmap alone should do the trick):
```haskell
-- I made up a random function name because I'm not quite sure what it does (:
makeJobScheduleValue s = object ["cron" .= J.cron s, "performFnArg" .= J.performFnArg s, "options" .= J.sheduleOptions s]
maybeJobSchedule =makeJobScheduleValue <$> J.schedule job
```

Reply:
Nice!
We could also break that object into multiple lines to make it more readable.

Reply:
I like it, thanks!